modules/ducktests/tests/checks/utils/check_marks.py

```diff
@@ -0,0 +1,240 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Checks custom parametrizers.
+"""
+
+import itertools
+from unittest.mock import Mock
+
+import pytest
+from ducktape.mark import parametrized, parametrize, matrix, ignore
+from ducktape.mark.mark_expander import MarkedFunctionExpander
+
+from ignitetest.utils._mark import IgniteVersionParametrize, ignite_versions, version_if
+from ignitetest.utils.version import IgniteVersion, V_2_8_0, V_2_8_1, V_2_7_6, DEV_BRANCH
+
+
+def expand_function(*, func, sess_ctx):
+    """
+    Inject parameters into function and generate context list.
+    """
+    assert parametrized(func)
+    assert next(filter(lambda x: isinstance(x, IgniteVersionParametrize), func.marks), None)
+
+    return MarkedFunctionExpander(session_context=sess_ctx, function=func).expand()
+
+
+def mock_session_ctx(*, global_args=None):
+    """
+    Create mock of session context.
+    """
+    sess_ctx = Mock()
+    sess_ctx.globals = global_args if global_args else {}
+
+    return sess_ctx
+
+
+class CheckIgniteVersions:
+    """
+    Checks @ignite_version parametrization.
+    """
+    single_params = itertools.product(
+        [[str(V_2_8_1)], [str(V_2_8_1), str(DEV_BRANCH)]],
+        [{}, {'ignite_versions': 'dev'}, {'ignite_versions': ['2.8.1', 'dev']}]
+    )
+
+    @pytest.mark.parametrize(
+        ['versions', 'global_args'],
+        list(map(lambda x: pytest.param(x[0], x[1]), single_params))
```

Comment:
Here and below, `list` is non-required for `parameterize` as it works well with generators. Additional converting to list increase code mess. Let's avoid it as much as possible.

Reply:
Ok